library/std/src/sys/pal/wasi/thread.rs

```diff
@@ -173,8 +173,7 @@ impl Thread {
         cfg_if::cfg_if! {
             if #[cfg(target_feature = "atomics")] {
                 unsafe {
-                    let ret = libc::pthread_join(self.id, ptr::null_mut());
-                    mem::forget(self);
+                    let ret = libc::pthread_join(mem::ManuallyDrop::new(self).id, ptr::null_mut());
```

Comment:
Split this into 2 lines with a `let id = ...`.

Reply:
Sure! I have done this for other platforms, but for some reason missed this one.